Add Svelte wrappers

metonym opened the pull request:
This feature implements `@carbon/charts` in Svelte to serve as a companion library for [carbon-components-svelte](https://github.com/IBM/carbon-components-svelte).

### Updates
- add `svelte` package under `packages/`
- add wrappers for Svelte
- use fixtures from `@carbon/charts/demo/data` to generate Storybook stories
- adjust build script for library to copy `src` to `dist` for Svelte component entry

### Demo screenshot or recording

Screenshot of the Storybook.

<img width="1534" alt="Screen Shot 2020-04-10 at 9 55 40 AM" src="https://user-images.githubusercontent.com/10718366/79008089-796bf480-7b11-11ea-8ee9-e2a7b17f3ace.png">


### Review checklist (for reviewers only)
- [ ] Demos all features
- [ ] Documented/annotated
- [ ] Matches UI/UX specs
- [ ] Meets the [code style guide](https://github.com/IBM/carbon-charts/wiki/Code-style-guide)
- [ ] Accessible
- [ ] Mobile first (responsive)
- [ ] RTL support (bidirectional text)
- [ ] Performant (limited bloat)

theiliad commented on packages/svelte/.babelrc:
```diff
@@ -0,0 +1,4 @@
+{
+	"presets": [["env", { "modules": false }]],
```
we should break this in multiple lines

theiliad commented on packages/svelte/README.md:
```diff
@@ -0,0 +1,42 @@
+
+## Codesandbox examples
+
+TBA
```
I've updated the READMEs for packages. We should do the same here

metonym replied:
I've filled in the Codesandbox examples sections in the README.

However, the "Step-by-step" section is still `TBA`. Should I add in a few steps for basic usage?

theiliad replied:
hmmm... yeah maybe at least in the README just add that

theiliad commented on packages/svelte/rollup.config.js:
```diff
@@ -0,0 +1,28 @@
+				"@carbon/charts": "Charts"
+			}
+		},
+		plugins: [svelte(), resolve(), terser()],
```
we should break this in multiple lines

theiliad commented on packages/svelte/package.json:
```diff
@@ -0,0 +1,83 @@
+		"sass": "1.25.0",
+		"sass-loader": "8.0.0",
+		"style-loader": "0.19.0",
+		"svelte": "^3.20.x",
```
```suggestion
		"svelte": "3.20.x",
```

theiliad commented on packages/svelte/package.json:
```diff
@@ -0,0 +1,83 @@
+		"sass-loader": "8.0.0",
+		"style-loader": "0.19.0",
+		"svelte": "^3.20.x",
+		"svelte-loader": "^2.13.6",
```
```suggestion
		"svelte-loader": "2.13.6",
```

theiliad commented on packages/svelte/package.json:
```diff
@@ -0,0 +1,83 @@
+	"devDependencies": {
+		"@storybook/addon-knobs": "5.3.12",
+		"@storybook/addon-options": "5.3.12",
+		"@storybook/svelte": "^5.3.18",
```
```suggestion
		"@storybook/svelte": "5.3.18",
```

ninja511 commented:
"svelte": "^3.20.x" is the only open source of relevance, and it's MIT, which is favorable to IBM. I just need to add new AWSOM approval for it. In the mean time, you can go ahead and proceed. Thanks.

theiliad commented:
Couple of points @metonym:
- would you be able to add codesandbox demos to the svelte stories? (you should be able to reference the vanilla/react stories on how they're being done, and here's a svelte codesandbox sample app https://codesandbox.io/s/svelte)
- I added a netlify app to build svelte deploy previews, should be good to go as long as you've setup similar NPM scripts in your `package.json` as the other packages
- we need to add a piece about the ownership of this work in terms of maintenance. I'm thinking of a `maintainers` field within `package.json` and a piece in the README pointing people towards you. I'll write up something for the README soon

metonym commented:
@theiliad 

1. Added CodeSandbox demos for Svelte
2. The npm scripts are identical to those of the react package
3. Added maintainers, contributors to `svelte/package.json`

theiliad commented on packages/svelte/package.json:
```diff
@@ -0,0 +1,97 @@
+{
+	"name": "@carbon/charts-svelte",
+	"version": "0.30.10",
```
```suggestion
	"version": "0.30.18",
```
